### PR TITLE
fix TestAccFirebaseWebApp_firebaseWebAppFull

### PR DESCRIPTION
**Pull request by shuyama1**
fixes https://github.com/hashicorp/terraform-provider-google/issues/16164

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by shuyama1** [comment]
test skipped in VCR. Running tests on TC: https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleBeta_MmUpstreamTesting_GOOGLEBETA_PACKAGE_FIREBASE/55335?hideTestsFromDependencies=false&hideProblemsFromDependencies=false

File: mmv1/third_party/terraform/services/firebase/resource_firebase_web_app_test.go.erb
Comment (rainshen49) [inline mmv1/third_party/terraform/services/firebase/resource_firebase_web_app_test.go.erb:1]: There is a bug in the API. Just to unblock this test, could you change the two instances of`browser_key_restrictions` to `allowed_referrers = []`?

**Comment by shuyama1** [comment]
new TC run: https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleBeta_MmUpstreamTesting_GOOGLEBETA_PACKAGE_FIREBASE/55340

**Comment by shuyama1** [comment]
new failure is 
```
vcr_utils.go:152: Step 1/4 error: After applying this test step, the plan was not empty.
        stdout:
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        Terraform will perform the following actions:
          # google_apikeys_key.key1 will be updated in-place
          ~ resource "google_apikeys_key" "key1" {
                id           = "projects/ci-test-project-188019/locations/global/keys/tf-test-api-key1csojhfm2gn"
                name         = "tf-test-api-key1csojhfm2gn"
                # (4 unchanged attributes hidden)
              ~ restrictions {
                  + browser_key_restrictions {
                      + allowed_referrers = [
                          + "",
                        ]
                    }
                }
            }
          # google_apikeys_key.key2 will be updated in-place
          ~ resource "google_apikeys_key" "key2" {
                id           = "projects/ci-test-project-188019/locations/global/keys/tf-test-api-key2csojhfm2gn"
                name         = "tf-test-api-key2csojhfm2gn"
                # (4 unchanged attributes hidden)
              ~ restrictions {
                  + browser_key_restrictions {
                      + allowed_referrers = [
                          + "",
                        ]
                    }
                }
            }
        Plan: 0 to add, 2 to change, 0 to destroy.
```
Looks like the API doesn't return back the empty list

**Comment by shuyama1** [comment]
TC: https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleBeta_MmUpstreamTesting_GOOGLEBETA_PACKAGE_FIREBASE/55808?buildTab=tests

**Review by NickElliot — COMMENTED** [review]
`TestAccFirebaseWebApp_firebaseWebAppCustomApiKeyExample` appears to be passing while still having the removed block in this test using the wildcard, do we know why the scenarios are different? I assume it's the b/310012020 issue, but if the issue is still present in the provider should the test highlighting it be temporarily fixed?

**Comment by rainshen49** [comment]
The API only checks the the allowed referrers of the API key when trying to update the `WebApp` entity. On creation, the API key is accepted as-is, which is why `TestAccFirebaseWebApp_firebaseWebAppCustomApiKeyExample` passed. Therefore, if the user just follows the example code in `TestAccFirebaseWebApp_firebaseWebAppCustomApiKeyExample`, `terraform apply` will succeed.

**Comment by NickElliot** [comment]
Sorry I kept editing my above comment so I think the specific thing I'm asking got missed -- the failure posted above seemed to suggest it was being hit before the update case was attempted (i.e. step 1/4 plan was not empty), so all that happened was the initial creation, and the only other difference is the presence of the data.google_firebase_web_app_config source.

**Comment by rainshen49** [comment]
There are two separate failures.

1.`TestAccFirebaseWebApp_firebaseWebAppFull` updates the `api_key_id` field of `WebApp` with a mismatched `allowed_referrers`. `TestAccFirebaseWebApp_firebaseWebAppCustomApiKeyExample` only creates the `WebApp` with a custom API key, but doesn't update, so it doesn't hit the issue. 
2. To fix `TestAccFirebaseWebApp_firebaseWebAppFull`, I suggested using an empty list for `allowed_referrers`, but it turns out the `google_apikeys_key` resource doesn't distinguish between empty list and unset `browser_key_restrictions`, thus the permadiff in https://github.com/GoogleCloudPlatform/magic-modules/pull/9251#issuecomment-1804823336. You're right that the creation of both `google_apikeys_key`s happens before the creation of `WebApp`.

Therefore, the only option at the moment is to make an unrestricted API key for testing updating the `api_key_id` field of `WebApp`.

**Review by NickElliot — APPROVED** [review]
Ah ok I'm seeing now, the perma-diff happened with an empty field as opposed to a wildcard. Since as you said the point of the test here is to verify the ability to update the API key, LGTM

**Pull request merged**